File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/adapter/WanSystemAdapter.java
package com.yyxnb.module_wanandroid.adapter;

import com.yyxnb.adapter.BaseAdapter;
import com.yyxnb.adapter.BaseViewHolder;
import com.yyxnb.common.ToastUtils;
import com.yyxnb.module_wanandroid.R;
import com.yyxnb.module_wanandroid.bean.WanClassifyBean;
import com.yyxnb.module_wanandroid.bean.WanSystemBean;
import com.yyxnb.view.text.FlowlayoutTags;

import java.util.ArrayList;
import java.util.List;

public class WanSystemAdapter extends BaseAdapter<WanSystemBean> {

    public WanSystemAdapter() {
        super(R.layout.item_wan_tags_layout);
    }

    @Override
    protected void bind(BaseViewHolder holder, WanSystemBean bean, int position) {
        holder.setText(R.id.tvTitle, bean.name);
        FlowlayoutTags mFlowlayout = holder.getView(R.id.mFlowlayout);
        List<String> tags = new ArrayList<>();
        for (WanClassifyBean classifyBean : bean.children) {
            tags.add(classifyBean.name);
        }

        mFlowlayout.setTags(tags);

        mFlowlayout.setOnTagClickListener(new FlowlayoutTags.OnTagClickListener() {
            @Override
            public void onTagClick(String tag) {
                ToastUtils.normal(tag);
            }
        });
    }
}

File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/adapter/WanSystemAdapter.kt
package com.yyxnb.module_wanandroid.adapter

import com.yyxnb.adapter.BaseAdapter
import com.yyxnb.adapter.BaseViewHolder
import com.yyxnb.common.CommonManager.toast
import com.yyxnb.module_wanandroid.R
import com.yyxnb.module_wanandroid.bean.WanSystemBean
import com.yyxnb.view.text.FlowlayoutTags
import java.util.*

class WanSystemAdapter : BaseAdapter<WanSystemBean>(R.layout.item_wan_tags_layout) {

    protected override fun bind(holder: BaseViewHolder, bean: WanSystemBean, position: Int) {
        holder.setText(R.id.tvTitle, bean.name)
        val mFlowlayout: FlowlayoutTags = holder.getView(R.id.mFlowlayout)
        val tags: MutableList<String> = ArrayList()
        for (t in bean.children!!) {
            tags.add(t.name)
        }
        mFlowlayout.setTags(tags)
        mFlowlayout.setOnTagClickListener { tag -> toast(tag) }
    }
}